examples/infer/parallel-prefix.desc

```diff
@@ -0,0 +1,247 @@
+                &uniq *view_exclusive_prefix, &uniq *b_flag, &uniq *b_agg, &uniq *b_prefix {
+                    if curr_block_id_x > 0 {
+                        let mut not_done = true;
+                        let mut endIndex = curr_block_id_x - 5;
+                        if endIndex < 0 {
+                            endIndex = 0
+                        };
+                        // TODO implement !done
+                        while not_done {
+                            *t_exclusive_prefix = 0;
+                            let mut i = 1;
+                            let mut flag = 0;
+                            let mut agg = 0;
+                            let mut prefix = 0;
+                            let mut not_break_loop = true;
+                            while i <= 5 && curr_block_id_x - i >= 0 && not_break_loop {
+                                let raw_ptr_flag = to_raw_ptr(&uniq *t_flag);
+                                let raw_ptr_flag_offset = offset_raw_ptr::<i32>(raw_ptr_flag, 0 - i);
+
+                                let raw_ptr_agg = to_raw_ptr(&uniq *t_agg);
+                                let raw_ptr_agg_offset = offset_raw_ptr::<i32>(raw_ptr_agg, 0 - i);
+
+                                let raw_ptr_prefix = to_raw_ptr(&uniq *t_prefix);
+                                let raw_ptr_prefix_offset = offset_raw_ptr::<i32>(raw_ptr_prefix, 0 - i);
+
+                                unsafe {
+                                    flag = *raw_ptr_flag_offset;
+                                    agg = *raw_ptr_agg_offset;
+                                    prefix = *raw_ptr_prefix_offset
+                                };
+
+                                if flag == 0 {
+                                    not_break_loop = false
+                                };
+                                if flag == 1 {
+                                    *t_exclusive_prefix = *t_exclusive_prefix + agg
+                                };
+                                if flag == 2 {
+                                    *t_exclusive_prefix = *t_exclusive_prefix + prefix;
+                                    not_break_loop = false;
+                                    not_done = false
+                                };
+                                i = i - 1
+                            }
+                        }
+                    }
+                };
+                parfor thread in split_thread_grp::<1>(block).0
+                with t_r, t_prefix, t_exclusive_prefix from &uniq *result_view, &uniq *b_prefix, &shrd *view_exclusive_prefix  {
+                    *t_prefix = *t_exclusive_prefix + *t_r
+                };
+                // Cheesy Threadfence
+                parfor thread in split_thread_grp::<1>(block).0
+                with t_flag from &uniq *b_flag {
+                    *t_flag = 2
+                };
+
+                // upsweep::<b_size, t_size>(block, &uniq shared_input)
```

Comment:
**This could be due to my changes!**

This would create the following code:

```
              [t_size, d, $r_281,
               b_size] __device__(descend::i32 *const arr_ref) -> void {
                {
                  for (std::size_t d = 1024; d > 0; d = d / 2) {

                    if (threadIdx.x < d) {
                      (&(*arr_ref))[(((threadIdx.x - 0) * (2048 / d)) +
                                     ((2048 / d) - 1))] =
                          (&(*arr_ref))[(((threadIdx.x - 0) * (2048 / d)) +
                                         ((2048 / d) - 1))] +
                          (&(*arr_ref))[(((threadIdx.x - 0) * (2048 / d)) +
                                         ((1024 / d) - 1))];
                    }
                    __syncthreads();
                  }
                }
              }(shared_input);
```

Reply:
Same goes for downsweep